### ISSUE TITLE
Problems when run sam deploy. EC2 Error Code: RequestLimitExceeded.

### ISSUE THREAD

**Issue by thinhlx1993** [aws/serverless-application-model#2007]
**Description:**

I have a Template.yaml and deploy around 130 lambdas and point it into API Gateways. I have run sam build, sam package before it. But when I executing sam deploy resource raise the exception errors 

> Resource handler returned message: "Your request has been throttled by EC2, please make sure you have enough  API rate limit. EC2 Error Code: RequestLimitExceeded.  EC2 Error Message: Request limit exceeded. (Service: Lambda, Status Code: 400, Request ID: 86ac65b9-7bd8-406d-82d3-8fd299108393, Extended Request ID: null)" (RequestToken: a3150497-ff39-bb23-0419-c1c95d87957e, HandlerErrorCode: InvalidRequest)

I don't know what wrong in my template.yaml

[template.txt](https://github.com/aws/serverless-application-model/files/6408813/template.txt)

**Comment by elbayaaa** [comment]
Thanks a lot @thinhlx1993, What happened when you retired the deployment?

When I tried to `sam deploy` your template, I didn't get throttle, instead I got the error

```
 Invalid Serverless Application Specification document. Number of errors found: 1. Resource with id [HBApiGatewayApiProduction] is invalid. Unable to add Cors configuration because 'AllowCredentials' can not be true when 'AllowOrigin' is "'*'" or not set
```

**Comment by elbayaaa** [comment]
TL;DR Make all of your lambda functions DependsOn the first function.

Diving more deep into your template, the issue is that it has a large number of lambda functions (around 130 lambdas) **inside a VPC**. When you connect a function to a VPC, Lambda makes API calls to EC2 to create an elastic network interface (ENI) for each subnet in your function's VPC configuration (check [this](https://docs.aws.amazon.com/lambda/latest/dg/configuration-vpc.html) doc). With this large number of Lambda functions created in parallel you are hitting the request limit and getting throttled. 

One work around is to use the [DependsOn](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-dependson.html) attribute to make your functions depend on each other and slow down the creation and not get throttled.

Also, Multiple functions connected to the same subnets **share network interfaces**. So, as all of your functions connects to the same subnets (subnet-020ce2dc3a177c52e & subnet-120df95f) I think if you made all of them DependsOn the first one, then only one ENI is created and shared and no throttles will happen.

I'm closing this for now. Please reopen if the workaround didn't work for you or if you have any further questions.
